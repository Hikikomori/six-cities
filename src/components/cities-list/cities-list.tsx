import {connect} from "react-redux";

import {Operation as baseOperation} from "../../reducer/base/base";

const {setCity} = baseOperation;

interface Props {
  activeCity: string,
  cities: Array<string>,
  setCity: (city: string) => void
}

const CitiesList: React.FC<Props> = (props) => {
  const {
    activeCity,
    cities,
    setCity} = props;

  return <div className="cities tabs">
    <section className="locations container">
      <ul className="locations__list tabs__list">
        {cities.map((city, id) => {
          return <li className="locations__item" key={id}>
            <a className={`locations__item-link tabs__item ${city === activeCity ? `tabs__item--active` : ``}`} href="#" onClick={(evt) => {
              evt.preventDefault();
              setCity(city);
            }}>
              <span>{city}</span>
            </a>
          </li>;
        })}
      </ul>
    </section>
  </div>;
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  activeCity: state.BASE.city,
  cities: state.BASE.cities,
});

const mapDispatchToProps = {
  setCity
};


export {CitiesList};

export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
