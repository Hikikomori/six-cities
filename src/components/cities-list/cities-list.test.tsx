import * as renderer from 'react-test-renderer';
import {CitiesList} from './cities-list';

it(`Cities List correctly renders`, () => {
  const tree = renderer
    .create(<CitiesList
      activeCity={``}
      cities={[`Amsterdam`]}
      setCity={jest.fn()}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
