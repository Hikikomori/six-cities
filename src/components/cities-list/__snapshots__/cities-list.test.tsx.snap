// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities List correctly renders 1`] = `
<div
  className="cities tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
